drivers/ksfilter/ks/swenum.c

```diff
@@ -2104,6 +2111,14 @@ KsServiceBusEnumPnpRequest(
             /* complete pending irps */
             KspCompletePendingIrps(DeviceEntry, STATUS_DEVICE_REMOVED);
 
+            /* free device extension */
+            FreeItem(ChildDeviceExtension);
+
+            /* FIXME this should only be done while installing the drivers */
```

Comment:
FYI, the right thing to do is to never delete PDOs in the first place, unless their parent FDO is removed.
See item 4 in https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/removing-a-device-in-a-bus-driver